lib/kube/proxy/watcher.go

```diff
@@ -174,10 +173,12 @@ func (m *monitoredKubeClusters) setResources(clusters types.KubeClusters) {
 	m.resources = clusters
 }
 
-func (m *monitoredKubeClusters) get() types.ResourcesWithLabelsMap {
+func (m *monitoredKubeClusters) get() map[string]types.KubeCluster {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	return append(m.static, m.resources...).AsResources().ToMap()
+	return services.ToMap(append(m.static, m.resources...), func(cluster types.KubeCluster) string {
+		return cluster.GetName()
+	})
```

Comment:
Pretty sure this works:

```suggestion
	return services.ToMap(append(m.static, m.resources...), types.KubeCluster.GetName)
```